### PR TITLE
Dev/user auth

### PR DESCRIPTION
**Pull request by Mr-Monwe**
Closing out user authentication using Google ReCaptcha API

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -1,23 +1,21 @@
...
+
+  <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
+  <!--
       manifest.json provides metadata used when your web app is installed on a
       user's mobile device or desktop. See https://developers.google.com/web/fundamentals/web-app-manifest/
     -->
-    <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
-    <!--
+  <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
+  <!--
```
Comment (jordantran22) [inline public/index.html:18]: I don't understand what is going on here, did you change anything? Looks like you deleted and repasted it for some reason.

File: public/index.html [hunk 2]
```diff
@@ -26,21 +24,22 @@
...
+    integrity="sha512-XQoYMqMTK8LvdxXYG3nZ448hOEQiglfqkJs1NOQV44cWnUrBc8PkAOcXy20w0vlaXaVUearIOBhiXZ5V3ynxwA=="
+    crossorigin=""></script>
+
+  <title>Foody</title>
+</head>
+
+<body>
+  <noscript>You need to enable JavaScript to run this app.</noscript>
+  <div id="root"></div>
+  <!--
```
Comment (jordantran22) [inline public/index.html:42]: Same thing here as well, anything changed?

File: src/components/Register.js [hunk 3]
```diff
@@ -76,53 +53,41 @@ const Register = ({background}) => {
...
+                        <input className="register-form-text-input" type="text" placeholder="Email" value={email} onChange={((e) => setEmail(e.target.value))}></input>
+                        <label> Enter a username </label>
+                        {invalidUsername ? <label className="invalid">Error: Username is already in use with another account. Try a different username!</label> : <label></label>}
+                        <input className="register-form-text-input" type="text" placeholder="Username" value={username} onChange={((e) => setUsername(e.target.value))}></input>
+                        <label> Enter a password </label>
+                        <input className="register-form-text-input" type="password" placeholder="Password" value={password} onChange={((e) => setPassword(e.target.value))}></input> 
+                                <ReCAPTCHA
+                                    sitekey="6LcgeDcdAAAAAOOgGLtrOpC5CcxujIZMLtyd-h33"
+                                    onChange={() => setIsVerified(false)} 
+                                />
```
Comment (jordantran22) [inline src/components/Register.js]: I think there should be some input verification messages for the user if they do not have the captcha clicked, for example "You must verify you are a human, click the box!" if the user tries to create the account without verifying. Similar implementation for missing username or password.

**Comment by jordantran22** [comment]
I think we should not merge the package-lock.json file, it contains 22,000 LOC changes

File: public/index.html (hunk 1 above)
Comment (Mr-Monwe) [reply public/index.html:18]: I did not touch this page, might be my autoformatter that did this.

File: public/index.html (hunk 2 above)
Comment (Mr-Monwe) [reply public/index.html:42]: Again, I did not make any changes to this page.

**Comment by Mr-Monwe** [comment]
> I think we should not merge the package-lock.json file, it contains 22,000 LOC changes

https://stackoverflow.com/questions/44206782/do-i-commit-the-package-lock-json-file-created-by-npm-5

**Pull request closed without merging**